doc/development/developer_guide.md

```diff
@@ -1,23 +1,28 @@
 # Developer Guide
 
-Please read the general [developer guide for the Scala projects][dev-guide].
+## S3 Write Path Validation
 
-## Integration Tests
+When using S3 storage as intermediate layer, we generate a S3 bucket path for intermediate data. The generated path is checked that it is empty before writing data.
+
+The path layout:
 
-The integration tests are run using [Docker][docker] containers. The tests use
-[exasol-testcontainers][exa-testcontainers] and
-[spark-testing-base][spark-testing-base].
+```
+userProvidedS3Bucket/
+└── uuid-sparkApplicationId
+    └── queryId
+```
+
+The generated intermediate write path `<UUID>-<SparkApplicationId>/<SparkQueryId>/` if validated that it is empty before write. And it is cleaned up after the write query finishes.
+
+## Integration Tests
 
-[docker]: https://www.docker.com/
-[exa-testcontainers]: https://github.com/exasol/exasol-testcontainers/
-[spark-testing-base]: https://github.com/holdenk/spark-testing-base
-[dev-guide]: https://github.com/exasol/import-export-udf-common-scala/blob/master/doc/development/developer_guide.md
+The integration tests are run using [Docker](https://www.docker.com) and [exasol-testcontainers](https://github.com/exasol/exasol-testcontainers/)
 
 ## Release
 
 Currently [release-droid](https://github.com/exasol/release-droid) has some troubles releasing the spark connector.
 
-* **Validation**: When using the local folder with `release-droid -l .` then validation failes with error message `NumberFormatException for input string: "v0"`.<br />
+* **Validation**: When using the local folder with `release-droid -l .` then validation fails with error message `NumberFormatException for input string: "v0"`.<br />
```

Comment:
This was fixed in release droid, please remove the note.

Reply:
Ahh great, I also had similar impression, but was not exactly sure.

Reply:
Removed whole section, since `relese_config.yml` file already contains language.